src/corg/coding-and-design-guidelines/code-review-guidelines.md

```diff
@@ -7,74 +7,235 @@ review:
 tree_item_index: 300
 toc: true
 ---
+
 Code reviews are a collaborative process between coders and reviewers — this is not a battle. The goal of the reviews is to improve the code quality by having several pairs of eyes on the code, for the benefit of all.
 
-## When Creating a Pull Request
+Before you read this page, make sure you've read the basic [Contribution Principals]({{page space='nxdoc' page='next/contributing-to-nuxeo'}}).
+
+These guidelines apply to the repositories of the [Nuxeo](https://github.com/nuxeo/) GitHub organization. Yet, some specific rules apply to the [Platform repositories](#platform-repositories), i.e. the ones maintained by the Platform team and built in a dedicated CI system.
+
+## When Creating a Pull Request (PR)
+
+### Use the Draft Feature
+
+Don’t hesitate to start with creating a draft PR. Then, you can ask for an early review if you feel it’s worth it, by using the GitHub [Reviewers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review) feature.
 
-### Follow the Coding Standards
+When the draft PR sounds mature enough and you want some people to actually review it, switch it to "Ready for review".
 
+You can, of course, create a PR and ask for reviews directly.
+In any case, make sure you read your own PR before submitting it to other people.
+
+### Follow the Coding and Design Guidelines
+
+Before you start writing some code, make sure you've read the [Coding and Design Guidelines]({{page page='coding-and-design-guidelines'}}).
 Important coding standards are described in the [Java Code Style]({{page page='java-code-style'}}) and [JavaScript Code Style]({{page page='javascript-code-style'}}) pages.
 
+It is important to understand that changes can be requested only because the code is not well formatted.
+
 ### Design for Easy Review
 
 The code must be designed for easy review and for future readers of the commits:
-- Don't mix cleanups/reformatting with actual code.
+
+- Don't mix cleanups/reformatting with actual code. For example, if you've changed a simple line of code in a class and re-formatted the whole file in a single commit, then your change is invisible in the diff, hidden by all the formatting changes.
+- Add the cleanup/format commit first.
 - Use several commits for unrelated work.
-- Use only one commit for related things. For example, don't have a separate "fix typo" commit if the typo fixed is in another commit of the Pull Request.
+- Use only one commit for related things. For example, don't have a separate "fix typo" commit if the typo fixed is in another commit of the PR.
+- Set a detailed commit description.
```

Comment:
I would love to have the 7 rules of great commit message applied:
https://chris.beams.io/posts/git-commit/